src/components/plots/abstractPlot.ts

```diff
@@ -445,6 +451,43 @@ export module Plot {
 
       return d3.selectAll(allSelections);
     }
+
+    /**
+     * Retrieves the closest PlotData to the specified x/y point within a specified value
+     *
+     * @param {number} xValue The x value to compare against
+     * @param {number} yValue The y value to compare against
+     * @param {number} withinValue The maximum distance the closest selection can be to the point (default = Infinity)
+     * @returns {PlotData} The closest plot data to the point within a specified value.  nulls and null selection returned otherwise
+     */
+    public getClosestData(xValue: number, yValue: number, withinValue: number = Infinity): PlotData {
+      var queryPoint = {x: xValue, y: yValue};
+
+      var closestDatum: any = null;
+      var closestSelection = d3.select();
+      var closestPixelPoint: Point = null;
+      var closestPointDistance = withinValue;
+
+      this._getDrawersInOrder().forEach((drawer) => {
+        drawer._getRenderArea().selectAll(drawer._getSelector()).each(function(datum: any, index: number) {
+          var selection = d3.select(this);
+          var pixelPoints = (drawer instanceof _Drawer.Line) ?
+                            (<any[]> datum).map((lineDatum, lineIndex) => drawer._getPixelPoint(lineDatum, lineIndex)) :
+                            [drawer._getPixelPoint(datum, index)];
```

Comment:
Notice the array format and thus _getPixelPoints only returns a single point but then gets wrapped into an array